crates/mun_hir/src/expr.rs

```diff
@@ -181,8 +192,50 @@ pub enum Statement {
 pub enum Literal {
     String(String),
     Bool(bool),
-    Int(i64),
-    Float(f64),
+    Int(LiteralInt),
+    Float(LiteralFloat),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum LiteralError {
+    /// We cannot parse the integer because its too large to fit in memory
+    IntTooLarge,
+
+    /// A lexer error occurred. This might happen if the literal is malformed (e.g. 0b01012)
+    LexerError,
+
+    /// Encountered an unknown suffix
+    InvalidIntSuffix(SmolStr),
+
+    /// Encountered an unknown suffix
+    InvalidFloatSuffix(SmolStr),
+
+    /// Trying to add floating point suffix to a literal that is not a floating point number
+    NonDecimalFloat(u32),
```

Comment:
Why for `float`, but not for `int`? E.g. `1.23u8`?

Reply:
Thats an `InvalidFloatSuffix`. 

It can only happen for floats because you can write `1f32` which is technically an integer. However because of that, you can also write `0x1f32` which is an invalid float.